ENYO-6316: Fix Marquee to start on focus when disabled

ryanjduffy opened the pull request:
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [X] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Focusing a disabled marquee element does not start marquee

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Remove the `!disabled` check in `MarqueeDecorator`

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
This matches `MarqueeController` which already allowed marqueeing when it was disabled. This regression is a hold over from the change to allow focusing disabled elements.

teckliew commented:
Not sure if we want to address this but when adding more than 1 line to `title` overflows out side of the dropdown button:
<img width="260" alt="twolinetitle" src="https://user-images.githubusercontent.com/8940009/67045594-7bfe0c00-f0e3-11e9-944e-f1dcf9c94a86.png">

ryanjduffy commented:
Pulled your commit in, @teck, and added another to clean up some other `disabled` references that were no longer relevant.

Of note, I removed `disabled` from `MarqueeDecorator` altogether because it is no longer used. This will cause it to appear as an API diff in our tooling but isn't since the prop is still forwarded on as noted in the prior docs. If anyone objects to this, we can add it back.

teckliew commented on packages/ui/Marquee/MarqueeDecorator.js:
```diff
@@ -377,7 +365,7 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 				this.cancelAnimation();
-			} else if (disabled && this.isHovered && marqueeOn === 'focus' && this.sync) {
+			} else if (this.isHovered && marqueeOn === 'focus' && this.sync) {
 				this.context.enter(this);
```
We might have to keep `disabled` for this `MarqueeController` sync logic, or there might be some logic needed for `handleLeave` in `MarqueeController` .

With `disabled` remove from here, `enter` is registered into the controller, but `leave` is never called when blurred.
